### PR TITLE
refactor: Remove unused code and update leaderboard generation

### PR DESCRIPTION
**Pull request by vante-dev**
- Remove unused code in the `route.ts` file
- Update the `leaderboard.ts` file to generate the leaderboard using data from the database instead of an external API
- Add a new file `leaderboard.tsx` to generate the leaderboard card image

**Please describe the changes this PR makes and why it should be merged:**

**Status and versioning classification:**

### CODE REVIEW

File: apps/bot/src/images/leaderboard.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import { ImageBuilder } from "phasebot/builders"
+
+import geistBold from "./fonts/geist-bold.otf"
+import geistMedium from "./fonts/geist-medium.otf"
+
+import { getOrdinal } from "~/lib/utils"
+
+interface dataProps {
```
Comment (mikaeladev) [inline apps/bot/src/images/leaderboard.tsx]: change this interface name to pascal case: `DataProps`

File: apps/bot/src/images/leaderboard.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  rank: number
+  target: number
+};
+
+
+interface LeaderboardCardProps {
+  data: dataProps[]
+}
+
+export const generateLeaderboardCard = async (props: LeaderboardCardProps) => {
```
Comment (mikaeladev) [inline apps/bot/src/images/leaderboard.tsx]: this doesn't need to be async

File: apps/bot/src/images/leaderboard.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  const leaderboardItems = await Promise.all(
```
Comment (mikaeladev) [inline apps/bot/src/images/leaderboard.tsx]: we can remove the `Promise.all` too

File: apps/bot/src/app/commands/(module)/level/leaderboard.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
-import { AttachmentBuilder } from "discord.js"
 import { BotSubcommandBuilder } from "phasebot/builders"
 
 import { ModuleId } from "@repo/utils/modules"
 
 import { BotErrorMessage } from "~/structures/BotError"
+import { db } from '~/lib/db.ts'
+import { generateLeaderboardCard } from '~/images/leaderboard.tsx'
```
Comment (mikaeladev) [inline apps/bot/src/app/commands/(module)/level/leaderboard.ts:7]: run the file through prettier to reorder these imports

**Review by mikaeladev — CHANGES_REQUESTED** [review]
looks good, just needs a few minor tweaks before it can be approved

File: apps/bot/src/app/commands/(module)/level/leaderboard.ts [hunk 5]
```diff
@@ -83,7 +83,7 @@ export default new BotSubcommandBuilder()
 
       const leaderBoardCard = await generateLeaderboardCard({
         data: response,
-      }).then((res) => res.toAttachment())
+      }).toAttachment()
```
Comment (notcharliee) [inline apps/bot/src/app/commands/(module)/level/leaderboard.ts:86]: you need to wrap the `generateLeaderboardCard` function call in brackets before you can run the `toAttachment` method

**Review by notcharliee — CHANGES_REQUESTED** [review]

**Pull request merged**